PR addressing Issue #214

Max-Bladen opened the pull request:
The `tune.spls()` function did not have any form of `BiocParallel` implementation which was quite misleading for users. As this function is likely to be more frequently used, this was a priority to resolve. 

The main body of the function was adjusted such that now the internal function `iter_keep.vals()` iterates over all pairs of `keepX`/`keepY`. Rather than doing it within each iteration, the function `DetermineOptimalKeepVals()` is called after all measure (`cor`, `RSS`) calculations. Impossible to iteratively build `measure.pred` using `bplapply()`.

Old function names and structure was preserved to the best of my ability, though some of it sorely needed a face lift

Max-Bladen committed:
Fix for Issue #214

fix: implemented BiocParallel functionality into `tune.spls()` via restructuring of the main portion of the function.

bplapply() iterates over all pairs of input keepX/Y so only reduces runtime for long lists of test.keepX/Y

Max-Bladen committed:
Fix for Issue #214

fix: implemented BiocParallel functionality into `tune.spls()` via restructuring of the main portion of the function.

bplapply() iterates over all pairs of input keepX/Y so only reduces runtime for long lists of test.keepX/Y

Max-Bladen committed:
Updated fix for Isssue #214

refactor: previous PR left duplicate function used for development. this is rectified and more comments are added

Max-Bladen committed:
Updated fix for Isssue #214

refactor: previous PR left duplicate function used for development. this is rectified and more comments are added

Max-Bladen committed:
Updated fix for Issue #214

test: added test cases for `tune.spls()`. These are just placeholders as more robust testing criteria needs to be established as well as this document will be fully updated as part of PR #206

Max-Bladen committed:
Updated fix for Issue #214

test: added test cases for `tune.spls()`. These are just placeholders as more robust testing criteria needs to be established as well as this document will be fully updated as part of PR #206

aljabadi commented:
Thanks, @Max-Bladen. Great work on spotting and addressing this.
I noticed that the checks for consistency of tune(method='spls') and tune.spls are dropped in unit tests. Was there a specific reason for that? Also, why was the RNG specifier removed in unit tests? 
Thanks

Max-Bladen commented:
Hi @aljabadi,

This is something I meant to ask you about but forgot. Due to my implementation of `bplapply()`, `set.seed()` is no longer affecting `tune.spls()`. Hence, calling the method twice on the same seed produced different results - the `tune()` and `tune.spls()` methods couldn't be equated. This is also why the tests currently just assess the class of the output and `.mixo_rng()` was removed. How can I make use of `bplapply()` without hindering the rng?

Max-Bladen committed:
Fix for Issue #214

tests: updated tests to assess actual values alongside RNG control of parallel function calls.

Max-Bladen committed:
Fix for Issue #214

tests: updated tests to assess actual values alongside RNG control of parallel function calls.

Max-Bladen committed:
Fix for Issue-214

refactor: replaced all calls to `cpus` to `BPPARAM` in `tune()`

Max-Bladen committed:
Fix for Issue-214

refactor: replaced all calls to `cpus` to `BPPARAM` in `tune()`

Max-Bladen committed:
Fix for Issue #214

docs: updated `tune.Rd` file with cpus -> BPPARAM shift

Max-Bladen committed:
Fix for Issue #214

docs: updated `tune.Rd` file with cpus -> BPPARAM shift

Max-Bladen commented:
Second set of commits on 13/12/2022 reflect initial set of commits, but replicated after the branch was rebased to master following PR #281.

Max-Bladen committed:
Merge branch 'issue-214' of https://github.com/mixOmicsTeam/mixOmics into issue-214
